### PR TITLE
Jupyterlab openmpp poc

### PR DESCRIPTION
**Pull request by vexingly**
# Description

**What your PR adds/fixes/removes**

# Tests / Quality Checks

## Are there breaking changes?
Ask yourself the next question;
- [ ] Do we want to maintain the previous image from which we had to do breaking changes from?

If no, then carry on. If yes, there is a breaking change and we **want to maintain the previous image** do the following
- [ ] Create a new branch for the current version (ex v1) based off the current master/main branch
- [ ]  Increment the tag in the CI for pushes to master/main (v1 to v2)
- [ ] Change the CI that on pushes to the newly created "v1" branch (the name of the newly created branch we want to maintain is) it will push to the ACR. 
## Automated Testing/build and deployment
- [ ] Does the image pass CI successfully (build, pass vulnerability scan, and pass automated test suite)?
- [ ] If new features are added (new image, new binary, etc), have new automated tests been added to cover these?
- [ ] If new features are added that require in-cluster testing (e.g. a new feature that needs to interact with kubernetes), have you added the `auto-deploy` tag to the PR before pushing in order to build and push the image to ACR so you can test it in cluster as a custom image?

## JupyterLab extensions

- [ ] Are all extensions "enabled" (`jupyter labextension list` from inside the notebook)?

## VS Code tests

- [ ] Does VS Code open?
- [ ] Can you install extensions?

## Code review

- [ ] Have you added the `auto-deploy` tag to your PR before your most recent push to this repo?  This causes CI to build the image and push to our ACR, letting reviewers access the built image without having to create it themselves
- [ ] Have you chosen a reviewer, attached them as a reviewer to this PR, and messaged them with the SHA-pinned image name for the final image to test on the **dev cluster** (e.g. `k8scc01covidacrdev.azurecr.io/jupyterlab-cpu:746d058e2f37e004da5ca483d121bfb9e0545f2b`)?

### CODE REVIEW

**Review by chuckbelisle — APPROVED** [review]
lgtm, this looks great. Thank you for the integration. We may have to rework once we move to multi-stage build.

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,7 @@ DEFAULT_NB_PREFIX := /notebook/username/notebookname
 
 .PHONY: clean .output generate-dockerfiles
 
+
```
Comment (Souheil-Yazji) [inline Makefile:49]: Remove

File: docker-bits/6_jupyterlab.Dockerfile [hunk 2]
```diff
@@ -121,6 +121,35 @@ RUN julia -e 'using Pkg; Pkg.add("LanguageServer")' && \
...
+# OpenM++ expects sqlite to be installed (not just libsqlite)
+RUN apt-get install --yes sqlite3 \
+    && wget -q https://github.com/openmpp/main/releases/download/v${OMPP_VERSION}/openmpp_debian_${OMPP_PKG_DATE}.tar.gz -O /tmp/ompp.tar.gz \
+    && echo "${SHA256ompp} /tmp/ompp.tar.gz" | sha256sum -c - \
+    && mkdir -p ${OMPP_INSTALL_DIR} \
+    && tar -xf /tmp/ompp.tar.gz -C ${OMPP_INSTALL_DIR} --strip-components=1
+    
+# Customize and rebuild omp-ui for jupyter-ompp-proxy install
+# issue with making a relative publicPath https://github.com/quasarframework/quasar/issues/8513
+RUN sed -i -e 's/history/hash/' ${OMPP_INSTALL_DIR}/ompp-ui/quasar.conf.js \
```
Comment (Souheil-Yazji) [inline docker-bits/6_jupyterlab.Dockerfile:142]: I don't think there's anything preventing this RUN and the above one from being merged, they also both pretain to OMPP. 1 less layer.

File: docker-bits/6_jupyterlab.Dockerfile [hunk 3]
```diff
@@ -121,6 +121,35 @@ RUN julia -e 'using Pkg; Pkg.add("LanguageServer")' && \
...
+    && sed -i -e "s/OMS_URL:.*''/OMS_URL: '.'/" ${OMPP_INSTALL_DIR}/ompp-ui/quasar.conf.js \
+    && npm install --prefix ${OMPP_INSTALL_DIR}/ompp-ui \
+    && npm run build --prefix ${OMPP_INSTALL_DIR}/ompp-ui \
+    && rm -r ${OMPP_INSTALL_DIR}/html \
+    && mv ${OMPP_INSTALL_DIR}/ompp-ui/dist/spa ${OMPP_INSTALL_DIR}/html \
+    && fix-permissions ${OMPP_INSTALL_DIR}
+
+COPY jupyter-ompp-proxy/ /opt/jupyter-ompp-proxy/
```
Comment (Souheil-Yazji) [inline docker-bits/6_jupyterlab.Dockerfile:150]: Move COPY to top

File: docker-bits/6_jupyterlab.Dockerfile [hunk 4]
```diff
@@ -121,6 +121,35 @@ RUN julia -e 'using Pkg; Pkg.add("LanguageServer")' && \
...
+RUN pip install /opt/jupyter-ompp-proxy/
```
Comment (Souheil-Yazji) [inline docker-bits/6_jupyterlab.Dockerfile:151]: Merge with L134/142

**Pull request merged**